javaslang/src/main/java/javaslang/collection/TreeMap.java

```diff
@@ -203,6 +207,105 @@ private TreeMap(RedBlackTree<Tuple2<K, V>> entries) {
     }
 
     /**
+     * Creates a HashMap of the given list of key-value pairs.
+     *
+     * @param <K> The key type
+     * @param <V> The value type
+     * @return A new Map containing the given entries
+     */
+    public static <K extends Comparable<? super K>, V> TreeMap<K, V> of(K k1, V v1, K k2, V v2) {
+        return of((Comparator<? super K> & Serializable) K::compareTo, k1, v1, k2, v2);
```

Comment:
I think this cast (`(Comparator<? super K> & Serializable`) is not Java 9 safe :/